pinot-core/src/main/java/org/apache/pinot/core/startree/operator/StarTreeFilterOperator.java

```diff
@@ -183,10 +186,12 @@ private BaseFilterOperator getFilterOperator() {
 
     // Add remaining predicates
     for (String remainingPredicateColumn : starTreeResult._remainingPredicateColumns) {
-      List<PredicateEvaluator> predicateEvaluators = _predicateEvaluatorsMap.get(remainingPredicateColumn);
+      List<CompositePredicate> predicateEvaluators = _predicateEvaluatorsMap.get(remainingPredicateColumn);
       DataSource dataSource = _starTreeV2.getDataSource(remainingPredicateColumn);
-      for (PredicateEvaluator predicateEvaluator : predicateEvaluators) {
-        childFilterOperators.add(FilterOperatorUtils.getLeafFilterOperator(predicateEvaluator, dataSource, numDocs));
+      for (CompositePredicate compositePredicate : predicateEvaluators) {
+        for (PredicateEvaluator predicateEvaluator : compositePredicate.getPredicateEvaluators()) {
```

Comment:
We need to check the filter type to build the child filter operator. Current code assume the type is `AND` which is not always the case